lib/Parse/ParseDecl.cpp

```diff
@@ -5646,8 +5662,9 @@ Parser::parseDeclSubscript(ParseDeclOptions Flags,
   Subscript->setGenericParams(GenericParams);
 
   // Pass the function signature to code completion.
-  if (SignatureHasCodeCompletion)
+  if (SignatureHasCodeCompletion && CodeCompletion) {
     CodeCompletion->setParsedDecl(Subscript);
```

Comment:
This is either a no-op or it was a really, really awful null pointer dereference. In any case, it's the safe thing to do.

Reply:
I think this is just defensive right now, but I hit it while in the process of changing the method and there is no reason that we shouldn't ever end up here during code-completion's first pass, so I left the check.